### ISSUE TITLE
Updating the 'name' field of a 'mongodbatlas_project' recreates a new Project

### ISSUE THREAD

**Issue by yannispgs** [mongodb/terraform-provider-mongodbatlas#1296]
### Terraform CLI and Terraform MongoDB Atlas Provider Version

```hcl
Terraform v1.5.2
on darwin_amd64
+ provider registry.terraform.io/mongodb/mongodbatlas v1.10.0
```

### Terraform Configuration File 

```hcl
resource "mongodbatlas_project" "main" {
  name   = var.project_name
  org_id = data.mongodbatlas_roles_org_id.main.org_id

  teams {
    team_id    = var.team_id
    role_names = ["GROUP_OWNER"]
  }
}
```

### Steps to Reproduce
1. Create/Import a MongoDB Project resource
2. Update the name of this Project
3. Run `terraform apply`

### Expected Behavior
Update the Project name in-place, without having to recreate a project (as in the UI).

### Actual Behavior
Terraform Plan indicates the Project will be recreated in order to update the Project's Name.

### Debug Output 
Terraform Cloud Plan output :
<img width="991" alt="Capture d’écran 2023-07-06 à 10 52 46" src="https://github.com/mongodb/terraform-provider-mongodbatlas/assets/67374171/b0eedbdf-13a2-4d87-89d6-8aa067facd4e">

### Additional Context
There is an [endpoint in the MongoDB Atlas API to update a Project Name](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Projects/operation/updateProject). I guess the Terraform provider should use that precise endpoint to update the Project Name.

It can highly impact a `terraform apply` command as the recreation will update the Project ID, leading to an update of all Resources linked to that Project.

**Comment by yannispgs** [comment]
Thanks for the quick fix @andreaangiolillo !

**Comment by andreaangiolillo** [comment]
Thanks! : ) The fix will be in the next release `v1.10.1` scheduled for this week. Have a lovely day!
